design/src/rfcs/rfc0004_retry_behavior.md

```diff
@@ -0,0 +1,147 @@
+RFC: Retry Behavior
+============================
+
+> Status: RFC
+
+For a summarized list of proposed changes, see the [Changes Checklist](#changes-checklist) section.
+
+It is not currently possible for users of the SDK to configure a client's retry behavior. This RFC establishes a method for users to set the number of retries to attempt when calling a service and would allow users to disable retries entirely.
+
+Terminology
+-----------
+
+- **Shared Config**: An `aws_config::Config` struct that is responsible for storing shared configuration data that is used across all services. This is not generated and lives in the `aws-config` crate.
+- **Service-specific Config**: A code-generated `Config` that has methods for setting service-specific configuration. Each `Config` is defined in the `config` module of its parent service. For example, the S3-specific config struct is `use`able from `aws_sdk_s3::config::Config` and re-exported as `aws_sdk_s3::Config`.
+
+Retry config
+--------------------
+
+Users can set the number of retries in several ways:
```

Comment:
Consider calling out that examples of these will follow. I was thinking, "it would be nice to have an example of that second bullet point," until I realized there was one later.

You might also consider making this an ordered list with the highest precedence first so that it reads like a fallback chain.

What is the default max attempts if none of these are set?

Reply:
Good idea. I'll call out the examples and update the order
The default (3) is specified [here](https://docs.aws.amazon.com/sdkref/latest/guide/setting-global-max_attempts.html)
I added a link to that page in the RFC